File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res=INT_MIN;
        int i=0;
        int j=height.size()-1;
        while(i<=j){
            int h=min(height[i],height[j]);
            res=max(res,h*(j-i));
            if(height[i]<height[j])
                i++;
            else
                j--;
        }
        return res;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans=float('-inf');
        i=0
        j=len(height)-1
        while(i<j):
            ans=max(ans,min(height[j],height[i])*(j-i))
            if(height[i]<height[j]):
                i+=1
            else:
                j-=1
        return ans